app/home/contribution/new/route.js

```diff
@@ -1,10 +1,21 @@
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
 
 export default class HomeContributionNewRoute extends Route {
   @service store;
 
   model() {
     return this.store.createRecord('contribution');
   }
+
+  @action
+  willTransition() {
+    const modelRecord = this.controller.get('model');
+
+    if (modelRecord.get('hasDirtyAttributes')) {
+      debugger;
+      modelRecord.destroyRecord();
+    }
+  }
```

Comment:
Please read this rule: https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-controller-access-in-routes.md, solution:
```suggestion
  resetController(controller, ...args) {
    super.resetController(controller, ...args);
    const modelRecord = controller.get('model');

    if (modelRecord.get('hasDirtyAttributes')) {
      modelRecord.destroyRecord();
    }
  }
```